java/code/src/com/suse/manager/webui/controllers/StatesAPI.java

```diff
@@ -657,13 +656,32 @@ else if (state.getPackageState() == PackageStates.REMOVED) {
         }
     }
 
+    /**
+     * Remove package state file for the given minion
+     *
+     * @param minion the minion
+     */
+    public static void removePackageState(MinionServer minion) {
+        Path baseDir = Paths.get(
+                SaltConstants.SUMA_STATE_FILES_ROOT_PATH, SaltConstants.SALT_PACKAGES_STATES_DIR);
+        Path filePath = baseDir.resolve(getPackagesSlsName(new MinionSummary(minion)));
+
+        try {
+            Files.deleteIfExists(filePath);
+        }
+        catch (IOException e) {
+            LOG.error(e.getMessage(), e);
+            throw new RuntimeException(e);
```

Comment:
Nitpick: Probably this won't be hit much but in this case, do we really want to cut the execution? Maybe just logging the error and continuing is a better option. We can even propagate the `IOException` and decide what to do with it from the calling code.